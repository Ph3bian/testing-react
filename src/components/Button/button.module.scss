@import "../../styles/variables";
.Button {
  width: 100%;
  font-size: 1rem;
  cursor: pointer;
  touch-action: manipulation;
  position: relative;
  color: $white;
  height: 2.25em;
  border-radius: 5px;
  font-family: "Poppins";
  border:0;
  background: $primary-color;
  &:hover{
    color: $white;
    background-color: transparentize($primary-color, 0.1);
    
  }
  &.neutral {
    border: $light-grey 1px solid;
    background-color:$light-grey;
    color: $primary-color;
  }
  &.primary {
    border: $primary-color 1px solid;
    background-color: $primary-color;
    color: $white;
  }
}
