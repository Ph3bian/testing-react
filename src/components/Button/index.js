import React from 'react';
import styles from './button.module.scss';

const Button = ({ type, children, onClick, isLoading, variant, title }) => (
  <button
    type={type ? type : 'button'}
    onClick={onClick}
    className={[styles['Button'], variant && styles[`${variant}`]].join(' ')}
    disabled={isLoading}
  >
    {isLoading ? 'Loading...' : title ? title : children}
  </button>
);
export default Button;
