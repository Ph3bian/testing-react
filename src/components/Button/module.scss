@import '../../styles/variables';
Button {
  width: 100%;
  font-size: 1rem;
  cursor: pointer;
  transition: all 0.2s;
  touch-action: manipulation;
  position: relative;
  padding: 1em 0.8em;
  line-height: 1.28rem;
  color: white;
  border-radius: 3px;
  background: $primary-color;
}
