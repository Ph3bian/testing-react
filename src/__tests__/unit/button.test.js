import React from 'react';
import ReactDOM from 'react-dom';
import { axe, toHaveNoViolations } from 'jest-axe';
import { render, fireEvent } from '@testing-library/react';
import Button from '../../components/Button';

describe('The <Button/> component', () => {
  expect.extend(toHaveNoViolations);

  const defaultProps = {
    type: 'button',
    onClick: jest.fn(),
    title: 'Submit',
  };

  const primaryButton = {
    type: 'button',
    onClick: jest.fn(),
    title: 'Submit',
    variant: 'primary',
  };
  test('Should render button correctly', () => {
    const { asFragment } = render(<Button {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should render button text correctly', () => {
    const { getByText } = render(<Button {...defaultProps} />);
    expect(getByText(/Submit/)).toBeInTheDocument();
  });

  test('Should call the onClick handler when it is provided', () => {
    const { getByText } = render(<Button {...defaultProps} />);
    fireEvent.click(getByText(defaultProps.title));
    expect(defaultProps.onClick).toHaveBeenCalled();
  });

  test("Should have class 'primary' if Button variant is primary", () => {
    const { getByText } = render(<Button {...primaryButton} />);
    expect(getByText(/Submit/)).toHaveAttribute('class');
    expect(getByText(/Submit/).classList).toHaveLength(2)
  });

  test('Should not fail any accessibility tests', async () => {
    const { container } = render(<Button {...primaryButton} />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
