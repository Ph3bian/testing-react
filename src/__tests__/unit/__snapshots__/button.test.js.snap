// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Button/> component Should render button correctly 1`] = `
<DocumentFragment>
  <button
    class="Button "
    type="button"
  >
    Submit
  </button>
</DocumentFragment>
`;
